bgrid_solo scripting and documentation update (#689)
NCAR/DART: models/bgrid_solo/readme.rst

variables are the gridded surface pressure, temperature, and u and v wind
components.

> The ``bgrid_solo`` directory has a ``work/workshop_setup.sh`` script that compiles

Reviewer: How about 
"The ``work`` directory has a ``workshop_setup.sh`` script that compiles"

and runs an example, producing a bgrid_solo netcdf file and perturbing it to get an
ensemble. This example is intended to demonstrate that the same process used for a
low-order model may be used for a much more complex model and generates output for
state-space or observation-space diagnostics.

Alternatively, the steps in the ``work/workshop_setup.sh`` script can be executed
indivudally on the command line. These steps are as follows:

1.  | ``./quickbuild.sh`` (or ``quickbuild.sh nompi`` if you are not building with mpi)

>     | This will create perfect_input.nc from the already available file
>       ``models/bgrid_solo/work/perfect_input.cdl``

Reviewer 2: ```suggestion
    | This will build all the DART executables and create perfect_input.nc from the cdl file
      ``models/bgrid_solo/work/perfect_input.cdl``
```

2.  | ``./perfect_model_obs``
    | Creates a synthetic observation sequence from a hindcast model

3.  | ``cp perfect_output.nc filter_input.nc``
    | Copies the output from perfect_model_obs program to the input file for filter

4.  | In ``work/input.nml``, set ``perturb_from_single_instance = .true.`` in the
      &filter_nml
    | This setting allows filter to generate an ensemble of perturbed ensemble member

>       restart files

Reviewer 2: 'allows filter' is a bit misleading. It is not allowing, it is _making_ filter do this. 

perturb_from_single_instance = .true will cause filter to perturb a single restart file to generate and ensemble.  Note this is not creating an ensemble of restart _files_.

5.  | ``./filter``

>     | Runs the assimilation program

Reviewer 2: 1-4 is a nice description of what the workshop_setup.sh script is doing. You are a bit loose with your language though. If you think it is necessary to add this new documentation, I think it is worth being precise - see the comments on the individual lines. 

It might be better to explain the 'why' of each step rather then explaining the command. e.g. why are you coping the perfect_output.nc to filter_input.nc?, why are you setting perturb_from_single instance?

Reviewer: I agree with Marlee that the steps to run the pmo experiment should be in readme.   I recommend going somewhat further and moving most of the text
from the beginning of workshop_setup to readme; everything from
'create_obs_sequence' through 'yet more matlab scripts'.  It should be combined with what Marlee has already added to readme, and some of it 
could use a little more (or less) context.
If we want the focus to be on the workshop_setup, then Marlee's text is good.
If we'd rather have the focus be on the steps, then it could be rearranged
to list the steps, and then note that they are all combined in workshop_setup.

In the section about Matlab scripts, are the model-space scripts in the local matlab directory?
And the 'observation space' scripts are in $DART/diagnostics/matlab?
Should there be an instruction about adding these directories to the Matlab path?

In pursuit of readability, when there's a list of descriptions, 
I find it helpful start each item on a new line.  
So, if the section about filter output stays in workshop_setup:
# 'filter' results in three files (at least): 
# + preassim.nc - the state of all ensemble members prior to the assimilation 
# + analysis.nc - the state of all ensemble members after the assimilation, and 
# + obs_seq.final - the ensemble members' estimates of the observations.
(I made a few corrections in addition to changing the formatting.)
If it's moved to readme, then use rst syntax to make it a list.

Author: Thanks for the suggestions Helen and Kevin. I will make sure to incporporate them and tidy up the docs a bit further once we decide if we want to keep the workshop_setup.sh script.

assimilate data from real-world sources is to compare the ensemble estimates of
the observation to your real-world data. You can estimate the bias and error of
the ensemble mean or gauge how many of the real-world observations are actually
being assimilated. These diagnostics are known as observation-space diagnostics.
DART provides the program ``obs_diag`` and MATLAB observation space diagnostics
for you to use to quickly assess the performance of your experiment. 

``obs_diag`` reads the ``obs_seq.final`` files and calculates several quantities,
such as the root-mean-squared error, bias, and spread, for an arbitrary number of
regions and levels. ``obs_diag`` outputs a netCDF file called ``obs_diag_output.nc``,
which can then be used in the DART MATLAB routines located in
``DART/diagnostics/matlab/``.

For more detail on obs_diag and the MATLAB diagnostics, see `PROGRAM obs_diag
(threed_sphere) <https://docs.dart.ucar.edu/en/latest/assimilation_code/programs
/obs_diag/threed_sphere/obs_diag.html>`_ and `MATLAB observation space diagnostics
<https://docs.dart.ucar.edu/en/latest/guide/matlab-observation-space.html>`_

> 

Reviewer 2: I know Kevin is reviewing this, but this is a definite change. Don't hardcode the html page as the link:

https://docs.dart.ucar.edu/en/latest/assimilation_code/programs/obs_diag/threed_sphere/obs_diag.html

Use the section headings or the custom anchors

Reviewer: Can we use a section heading on a different page?
Is that how custom anchors would be used?

Would it work (long term) to use relative path names?
E.g. ../../assimilation_code/programs/obs_diag/threed_sphere/obs_diag.html

Reviewer 2: so yes and no for the relative paths
you don't want ../../assimilation_code/programs/obs_diag/threed_sphere/obs_diag.html
with .html because you might not be building html. 
you can use :doc:../../assimilation_code/programs/obs_diag/threed_sphere/obs_diag (no extension) 
with the relative path. 

Your link will break if the .rst file is moved. vs. if you link to the anchor the link will still work if you rearrange the files. 

also you want to be really careful to not hardcode the version, e.g. 
https://docs.dart.ucar.edu/en/**latest**/assimilation_code/programs
/obs_diag/threed_sphere/obs_diag.html
because then the link takes you the hardcoded version, rather than the page within the version of the documentation you are looking at. 
e.g for this pull request the documentation is here https://dart-documentation--689.org.readthedocs.build/en/689/
but the link takes you to https://docs.dart.ucar.edu/en/latest/

I know we have all sorts of links of various flavours all over the documentation, but I think it is worth being careful in pull requests to not introduce
